--- a/setup.py	2025-01-23 19:54:30.894477198 -0700
+++ b/setup.py	2025-01-23 19:55:15.894461013 -0700
@@ -392,7 +392,7 @@
        (netcdf_lib_version > "4.4" and netcdf_lib_version < "4.5"):
         has_cdf5_format = True
 
-    has_parallel_support = check_has_parallel_support(inc_dirs)
+    has_parallel_support = False
     has_has_not = "has" if has_parallel_support else "does not have"
     print(f"netcdf lib {has_has_not} parallel functions")
 
