From 088f00fb9cbd933345a56cf79f862990b82b4357 Mon Sep 17 00:00:00 2001
From: Rocco Meli <r.meli@bluemail.ch>
Date: Fri, 4 Oct 2024 22:10:44 +0200
Subject: [PATCH] jsoncpp include

---
 gninasrc/lib/torch_model.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gninasrc/lib/torch_model.h b/gninasrc/lib/torch_model.h
index 5ea785ae..86f30f11 100644
--- a/gninasrc/lib/torch_model.h
+++ b/gninasrc/lib/torch_model.h
@@ -9,7 +9,7 @@
 #define SRC_LIB_TORCH_MODEL_H_
 
 #include <iostream>
-#include <jsoncpp/json/json.h>
+#include <json/json.h>
 #include <libmolgrid/atom_typer.h>
 #include <libmolgrid/grid_maker.h>
 #include <torch/script.h>
@@ -46,4 +46,4 @@ template <bool isCUDA> class TorchModel {
   std::shared_ptr<libmolgrid::AtomTyper> get_rec_typer() const { return rec_typer; }
 };
 
-#endif /* SRC_LIB_TORCH_MODEL_H_ */
\ No newline at end of file
+#endif /* SRC_LIB_TORCH_MODEL_H_ */
