diff --git a/tools/chapel-py/setup.py b/tools/chapel-py/setup.py
index 30c2708724..fd2fe54e12 100644
--- a/tools/chapel-py/setup.py
+++ b/tools/chapel-py/setup.py
@@ -25,6 +25,7 @@ import sys
 import glob

 chpl_home = str(os.getenv("CHPL_HOME"))
+chpl_lib_install = str(os.getenv("CHPL_SPACK_LIB_INSTALL"))
 chpl_printchplenv = os.path.join(chpl_home, "util", "printchplenv")
 chpl_variables_lines = (
     subprocess.check_output(
@@ -66,6 +67,7 @@ LDFLAGS += [
     "-L{}".format(chpl_lib_path),
     "-Wl,-rpath,{}".format(chpl_lib_path),
     "-lChplFrontendShared",
+    "-Wl,-rpath,{}".format(chpl_lib_install),
 ]

 if str(chpl_variables.get("CHPL_SANITIZE")) == "address":
